### PR TITLE
Improved the Titles tab on the Graphs subdialog

### LINKED ISSUE #8299: Improve the Titles tab on the graphs sub-dialogue
Here is the current Titles tab:

image

And here is the Titles tab for our Climatic > PICSA graphs.

image

Perhaps you can move that tab across and use it instead?  I like the option there to dictate the size of the titles.

I wonder whether the x-axis and y-axis titles go well there too - instead of on the x and y tabs?

Before that, please also add a 4th option for a tag.  This is also simply a labs(tag = "A") or "234" or anything, but usually short so make the place for it shorter.

 And perhaps a fifth, namely `New Legend Title` .  This is also reasonably short.  The R command It is labs(colour = "new_title", or fill = "new title") because the default legend title is the name of the variable specified in either fill or colour.  I found that it just ignores the result if wrong, so you could give both always?  So labs(colour="roger",fill="roger) seems to work.

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #8299 
@rdstern @lloyddewit , this PR improves the titles tabs in Plot sub-dialog

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@Vitalis95 this is looking very promising.  Just what I was hoping for.

a) Could you also tempt the Caption to be multi-line as we do in the PICSA climatic sub-dialogue.
b) Could you also add the tooltip from there that explains that \n can be used to provide a multiline caption.

c) A sutprise!
image

Here is an example, that gave me a small surprise.  The x and y axis sizes seemed to be affected by the changes in the font size for the title, or subtitle.  (I would like also to be able to choose their size, but perhaps not there!

d) There is something odd with your code and the defaults.  I get a very different size for the plot title and sub-title when I have the defaults of 20 and 15, compared with 20.1 and 15.1.  Could thise defaults be the actual ones as well as not linking to the x axis and y axis sizes.

e) Could the Tag and New Legend Title have checkboxes, default unchecked,  The text is only visible when checked.  Maybe the tag and also perhaps the Legend needs the Size too, and I am not sure the Legend title works yet?

**Comment by Vitalis95** [comment]
@rdstern have a look at it

**Review by rdstern — COMMENTED** [review]
@Vitalis95 this is looking good now.
a) When I first unchecked the tag it was as follows:
image

But when I returned to the sub-dialog the size of the tag had changed to 8?
b) In the resulting ggplot I am not sure all the components are the size givewn - sometimes it is given, and sometimes not, so it presumably uses the default.  I suggest anytime a component is used (i.e. not null) then you explicitely put the size into the R command.
c) I am not sure what the Fill is doing for the new legend?  Doesn't that provide a colour?  If so, then the type of control is not correct.

**Review by rdstern — CHANGES_REQUESTED** [review]
@Vitalis95 here is the graph after input of a title and sub-title and caption.
I think it is probably a single error, but it looks as though the y-axis and x-axis and legends title are the same size as the main title?

**Comment by Vitalis95** [comment]
@rdstern , have a look at it

**Review by rdstern — COMMENTED** [review]
@Vitalis95 this all seems to work well now.

My only problem is the funny 2 parameters for the legend.  This entry seems to indicate I am not the only one to be [confused](https://stackoverflow.com/questions/14622421/how-to-change-legend-title-in-ggplot).

I suggest you just have the Legend title control there  - and still use the fill parameter in the R code that you generate.
 Then we can merge.

**Comment by Vitalis95** [comment]
@rdstern , check it now

**Review by rdstern — DISMISSED** [review]
@Vitalis95 great.  @lloyddewit over to you

File: instat/sdgPlots.vb [hunk 1]
```diff
@@ -38,6 +38,11 @@ Public Class sdgPlots
     Public clsScaleColourViridisFunction As New RFunction
     Public clsScaleFillViridisFunction As New RFunction
     Private clsAnnotateFunction As New RFunction
+    Private clsPlotElementTitle As New RFunction
+    Private clsPlotElementSubTitle As New RFunction
+    Private clsPlotElementCaption As New RFunction
+    Private clsPlotElementTag As New RFunction
+    Private clsPlotLegendTitle As New RFunction
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:45]: Please follow naming conventions and rename these 5 variables to `clsPlotElementTitleFunction` etc.

File: instat/sdgPlots.vb [hunk 2]
```diff
@@ -193,6 +198,46 @@ Public Class sdgPlots
...
+        ucrNudCaptionSize.SetMinMax(0, Integer.MaxValue)
+        'ucrNudCaptionSize.SetRDefault(8)
+
+        ucrNudTagSize.SetParameter(New RParameter("size"))
+        ucrNudTagSize.SetMinMax(0, Integer.MaxValue)
+        ucrNudTagSize.SetRDefault(20)
+
+        ucrNudLegendSize.SetParameter(New RParameter("size"))
+        ucrNudLegendSize.SetMinMax(0, Integer.MaxValue)
+        'ucrNudLegendSize.SetRDefault(18)
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:222]: what's the reason for commenting out some of these lines? Delete these lines?

File: instat/sdgPlots.vb [hunk 3]
```diff
@@ -576,6 +621,9 @@ Public Class sdgPlots
         ucrChkAddColour.AddParameterPresentCondition(True, "scale_colour", True)
         ucrChkAddColour.AddParameterPresentCondition(False, "scale_colour", False)
 
+        ttCaptionTitle.SetToolTip(ucrInputGraphCaption.txtInput, "Type \n where you would like a new-line")
+
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:626]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 4]
```diff
@@ -685,6 +767,7 @@ Public Class sdgPlots
         ucrXAxis.SetRCodeForControl(bIsXAxis:=True, strNewAxisType:=GetAxisType(True), clsNewXYlabTitleFunction:=clsXLabFunction, clsNewXYScaleContinuousFunction:=clsXScalecontinuousFunction, clsNewXYScaleDateFunction:=clsXScaleDateFunction, clsNewBaseOperator:=clsBaseOperator, bReset:=bReset, bCloneIfNeeded:=True)
         ucrYAxis.SetRCodeForControl(bIsXAxis:=False, strNewAxisType:=GetAxisType(False), clsNewXYlabTitleFunction:=clsYLabFunction, clsNewXYScaleContinuousFunction:=clsYScalecontinuousFunction, clsNewBaseOperator:=clsBaseOperator, clsNewXYScaleDateFunction:=clsYScaleDateFunction, bReset:=bReset, bCloneIfNeeded:=True)
 
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:770]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 5]
```diff
@@ -998,6 +1091,7 @@ Public Class sdgPlots
     Private Sub AddRemoveXElementTitle()
         If clsXElementTitle.iParameterCount > 0 Then
             clsThemeFunction.AddParameter("axis.title.x", clsRFunctionParameter:=clsXElementTitle)
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1094]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 6]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
                 clsAnnotateFunction.RemoveParameterByName("label")
         End Select
     End Sub
+
+
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1255]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 7]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
...
         End Select
     End Sub
+
+
+
+    Private Sub ucrChkTag_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkTag.ControlValueChanged, ucrInputTag.ControlValueChanged, ucrNudTagSize.ControlValueChanged
+        If ucrChkTag.Checked AndAlso Not ucrInputTag.IsEmpty Then
+            clsLabsFunction.AddParameter("tag", Chr(34) & ucrInputTag.GetText & Chr(34))
+            clsThemeFunction.AddParameter("plot.tag", clsRFunctionParameter:=clsPlotElementTag)
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1260]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 8]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
...
+        Else
+            clsLabsFunction.RemoveParameterByName("tag")
+            clsThemeFunction.RemoveParameterByName("plot.tag")
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1264]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 9]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
...
+        End If
+    End Sub
+
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1268]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 10]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
...
+    Private Sub ucrChkNewLegend_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkNewLegend.ControlValueChanged, ucrInputLegendTitle.ControlValueChanged, ucrNudLegendSize.ControlValueChanged
+        If ucrChkNewLegend.Checked AndAlso Not ucrInputLegendTitle.IsEmpty Then
+            clsLabsFunction.AddParameter("colour", Chr(34) & ucrInputLegendTitle.GetText & Chr(34))
+            clsLabsFunction.AddParameter("fill", Chr(34) & ucrInputLegendTitle.GetText & Chr(34))
+            clsThemeFunction.AddParameter("legend.title", clsRFunctionParameter:=clsPlotLegendTitle)
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1274]:
```suggestion
```

File: instat/sdgPlots.vb [hunk 11]
```diff
@@ -1156,4 +1250,34 @@ Public Class sdgPlots
...
+        Else
+            clsLabsFunction.RemoveParameterByName("colour")
+            clsLabsFunction.RemoveParameterByName("fill")
+            clsThemeFunction.RemoveParameterByName("legend.title")
+
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:1279]:
```suggestion
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

File: instat/sdgPlots.vb (hunk 1 above)
Comment (lloyddewit) [reply instat/sdgPlots.vb:45]: You marked this as resolved but I don't see any related changes.
Please could you let me know the status? thanks

File: instat/sdgPlots.vb (hunk 2 above)
Comment (lloyddewit) [reply instat/sdgPlots.vb:222]: You marked this as resolved but I don't see any related changes.
Please could you let me know the status? thanks

**Comment by lloyddewit** [comment]
@Vitalis95 Thank you for the changes, I think there are still 2 unresolved comments above, please could you check?
thanks

File: instat/sdgPlots.vb (hunk 2 above)
Comment (Vitalis95) [reply instat/sdgPlots.vb:222]: > You marked this as resolved but I don't see any related changes. Please could you let me know the status? thanks

@lloyddewit hold on, I am resolving a conflict I have then push it. I will notify you

**Comment by Vitalis95** [comment]
@lloyddewit , it is fine now.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern There are some small changes since your last approval. If you can test/approve again, then we can merge, thanks

**Review by rdstern — APPROVED** [review]
Great.  Thanks

**Pull request merged**